### PR TITLE
Add showing number of new pages added from yesterday

### PR DESCRIPTION
**Pull request by sviatoslav-krupa**
**CHECK LIST**
- [ ] specs added
- [x] specs passed
- [x] code coverage >90%
- [ ] DB indexes added if needed
- [ ] seed.rb file is updated if needed
- [x] PR is reviewed manually again (to make sure you have 100% ready code)
- [x] PR doesn't have conflicts with main branch
- [ ] I've checked new feature as logged in and logged out user if needed
- [x] "let!" is used in specs
- [ ] screenshot of rails_best_practices and other tools added
- [x] screenshot of feature is added if task has UI

image

### CODE REVIEW

File: app/controllers/account/dashboard_controller.rb [hunk 1]
```diff
@@ -3,5 +3,18 @@ class Account::DashboardController < ApplicationController
...
   def index
     @chart_value = current_companies
+    @count_of_pages = todays_pages
+  end
+
+  private
+
+  def todays_pages
+    count = 0
+    current_companies.each do |company|
```
Comment (igorkasyanchuk) [inline app/controllers/account/dashboard_controller.rb]: what if you have 100000 companies in DB?

Use one line of code to count all pages for all companies!!!

File: app/controllers/account/dashboard_controller.rb [hunk 2]
```diff
@@ -3,5 +3,18 @@ class Account::DashboardController < ApplicationController
...
+      count += company.pages
+                      .where('created_at > ?', 1.day.ago)
```
Comment (igorkasyanchuk) [inline app/controllers/account/dashboard_controller.rb]: PLUS 

I expect this code to be a part of new class

CompanyPagesAnalytics.count_pages to return it

File: app/controllers/account/dashboard_controller.rb [hunk 3]
```diff
@@ -3,5 +3,18 @@ class Account::DashboardController < ApplicationController
 
   def index
     @chart_value = current_companies
+    @count_of_pages = todays_pages
+  end
+
+  private
+
+  def todays_pages
```
Comment (romaslmd) [inline app/controllers/account/dashboard_controller.rb]: make count on DB lvl, using ActiveRecord query

File: app/helpers/company_pages_analytics_helper.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+module CompanyPagesAnalyticsHelper
```
Comment (igorkasyanchuk) [inline app/helpers/company_pages_analytics_helper.rb]: implement as class

File: app/helpers/company_pages_analytics_helper.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+  def count_pages
```
Comment (igorkasyanchuk) [inline app/helpers/company_pages_analytics_helper.rb]: as class method

File: app/helpers/company_pages_analytics_helper.rb [hunk 6]
```diff
@@ -0,0 +1,8 @@
...
+    Page.where(
```
Comment (igorkasyanchuk) [inline app/helpers/company_pages_analytics_helper.rb]: ALL pages?
where is tenant?

Comment (sviatoslav-krupa) [reply app/helpers/company_pages_analytics_helper.rb]: Can I add has_many_though association to tenant model like this:

Tenant:
has_many :pages, through: :company

Company:
belongs_to :tenant
has_many :pages

Page:
belongs_to :tenant

And then, call @current_tenant.pages... in that method?

File: app/controllers/account/dashboard_controller.rb [hunk 7]
```diff
@@ -1,7 +1,10 @@
 class Account::DashboardController < ApplicationController
+  include CompanyPagesAnalyticsHelper
```
Comment (igorkasyanchuk) [inline app/controllers/account/dashboard_controller.rb]: CLASS!!! NOT MODULE

**Comment by igorkasyanchuk** [comment]
yes http://prntscr.com/i8ttdp

File: app/models/page.rb [hunk 8]
```diff
@@ -43,7 +43,8 @@ class Page < ApplicationRecord
   PENDING_TITLE    = 'pending'.freeze
 
   belongs_to :company
-  has_many   :comments, as: :commentable, dependent: :destroy
+  delegate :tenant, to: :company, allow_nil: true
```
Comment (romaslmd) [inline app/models/page.rb]: associations then delegations
keep ordering

File: app/lib/company_pages_analytics.rb [hunk 9]
```diff
@@ -0,0 +1,8 @@
+class CompanyPagesAnalytics
```
Comment (romaslmd) [inline app/lib/company_pages_analytics.rb]: a whole new class only for this count?

File: app/lib/company_pages_analytics.rb [hunk 10]
```diff
@@ -0,0 +1,8 @@
...
+  def self.count_pages(current_tenant)
+    current_tenant.pages.where(
```
Comment (romaslmd) [inline app/lib/company_pages_analytics.rb]:
```
current_tenant.pages.exclude_chrome_ext.in_period(1.day.ago).count
```

```
# app/models/page.rb

scope :exclude_chrome_ext, -> { where.not(page_type: CHROME_EXTENSION) }
scope :in_period, ->(period) { where('pages.created_at > ?', period) }
```

File: app/lib/company_pages_analytics.rb (hunk 9 above)
Comment (igorkasyanchuk) [reply app/lib/company_pages_analytics.rb]: yes :) I've asked it. It will get more methods in future

File: app/models/page.rb (hunk 8 above)
Comment (igorkasyanchuk) [reply app/models/page.rb]: yes, Roman right. you have already delegate in controller. DON"T make a mess in code. FIX IT in new PR

**Pull request merged**